x/wasm/client/testutil/query.go

```diff
@@ -0,0 +1,466 @@
+package testutil
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/gogo/protobuf/proto"
+	clitestutil "github.com/line/lbm-sdk/testutil/cli"
+	"github.com/line/lbm-sdk/types/query"
+	"github.com/line/lbm-sdk/x/wasm/client/cli"
+	"github.com/line/lbm-sdk/x/wasm/types"
+)
+
+func (s *IntegrationTestSuite) TestGetCmdListCode() {
+	val := s.network.Validators[0]
+
+	var args []string
+	cmd := cli.GetCmdListCode()
+	out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(args, s.queryCommonArgs()...))
+	s.Require().NoError(err)
+
+	codeID, err := strconv.ParseUint(s.codeID, 10, 64)
+	s.Require().NoError(err)
+
+	var codes types.QueryCodesResponse
+	s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &codes), out.String())
+	s.Require().GreaterOrEqual(1, len(codes.CodeInfos))
+	s.Require().Equal(codes.CodeInfos[0].CodeID, codeID)
+}
+
+func (s *IntegrationTestSuite) TestGetCmdListContractByCode() {
+	val := s.network.Validators[0]
+
+	testCases := map[string]struct {
+		args     []string
+		valid    bool
+		expected proto.Message
+	}{
+		"valid query": {
+			[]string{
+				s.codeID,
+			},
+			true,
+			&types.QueryContractsByCodeResponse{
+				Contracts:  []string{s.contractAddress},
+				Pagination: &query.PageResponse{},
+			},
+		},
+		"no codeID": {
+			[]string{},
+			false,
+			nil,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+
+		s.Run(name, func() {
+			cmd := cli.GetCmdListContractByCode()
+			out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(tc.args, s.queryCommonArgs()...))
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NoError(err)
+
+			var contracts types.QueryContractsByCodeResponse
+			s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &contracts), out.String())
+			s.Require().Equal(tc.expected, &contracts)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestGetCmdQueryCodeInfo() {
+	val := s.network.Validators[0]
+
+	codeID, err := strconv.ParseUint(s.codeID, 10, 64)
+	s.Require().NoError(err)
+	dataHash, err := hex.DecodeString("470C5B703A682F778B8B088D48169B8D6E43F7F44AC70316692CDBE69E6605E3")
```

Comment:
How about adding the generation logic of it instead of the literal?

Reply:
I think if then it is not unittest. Because this is the test that check if the result is expected data or not. So I think it's better to change the `dataHash` to `expectedDataHash`.

Reply:
And the generation logic a little complicate. And I want to check if the generation logic is right and not changed

Reply:
I see. So you mean this line only checks some basic constraints and it will be used for the test below. Then, it's fine.